include/devicetree/pwms.h

```diff
@@ -101,6 +101,79 @@ extern "C" {
  */
 #define DT_PWMS_LABEL(node_id) DT_PWMS_LABEL_BY_IDX(node_id, 0)
 
+/**
+ * @brief Get a pwm-controller node identifier from a pwms property at an index
+ *
+ * Example devicetree fragment:
+ *
+ *     pwm1: pwm-controller@... {
+ *             label = "PWM_1";
+ *     };
+ *
+ *     pwm2: pwm-controller@... {
+ *             label = "PWM_2";
+ *     };
+ *
+ *     n: node {
+ *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
+ *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_PWMS_CTLR_BY_IDX(DT_NODELABEL(n), 0) // "PWM_1"
+ *     DT_PWMS_CTLR_BY_IDX(DT_NODELABEL(n), 1) // "PWM_2"
+ *
+ * @param node_id node identifier for a node with a pwms property
+ * @param idx logical index into pwms property
+ * @return  The node identifier for the node referenced at index "idx"
```

Comment:
```suggestion
 * @return the node identifier for the PWM controller referenced at
 *        index "idx"
```

For consistency with DT_GPIO_CTLR_BY_IDX